web/html/src/manager/maintenance/calendar-edit.js

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable */
+'use strict';
+
+import React, {useState, useEffect, forwardRef, useImperativeHandle} from "react";
+import {Check} from "components/input/Check";
+import {Form} from "components/input/Form";
+import {Text} from "components/input/Text";
+import {Button} from "components/buttons";
+import {ModalButton} from "components/dialog/ModalButton";
+import {DangerDialog} from "components/dialog/DangerDialog";
+
+const MaintenanceCalendarEdit = forwardRef((props, ref) => {
+    const [calendarName, setCalendarName] = useState("");
+    const [calendarData, setCalendarData] = useState();
+    const [calendarDataText, setCalendarDataText] = useState("");
+    const [icalLoading, setIcalLoading] = useState(false);
+    const [strategy, setStrategy] = useState(false);
```

Comment:
`calendarName` and `strategy` also belong to the `model` object. To avoid duplication, you should define `model` with `setState` and use/assign properties inside the model directly.
Hint: You'll need to use `Object.assign()` to modify the `model` properly.

Reply:
Makes sense to me, I added a `model` state.